send-emails-from-a-service/main.bal

```diff
@@ -0,0 +1,173 @@
+            return <http:InternalServerError> {body: smtpClient.message()};
+        } 
+
+        ReservationRequest {
+            patient: {cardNo, ...patient},
+            doctor,
+            hospital,
+            hospital_id,
+            appointment_date
+        } = reservationRequest;
+
+        Appointment|http:ClientError appointment =
+                hospitalServicesEP->/[hospital_id]/categories/[category]/reserve.post({
+            patient,
+            doctor,
+            hospital,
+            appointment_date
+        });
+
+        if appointment !is Appointment {
+            log:printError("Appointment reservation failed", appointment);
+            if appointment is http:ClientRequestError {
+                return <http:NotFound> {body: string `unknown hospital, doctor, or category`};
+            }
+            return <http:InternalServerError> {body: appointment.message()};
+        }
+
+        int appointmentNumber = appointment.appointmentNumber;
+
+        PaymentSettlement paymentSettlement = {
+            appointmentNumber: appointmentNumber,
+            doctor: appointment.doctor,
+            patient: appointment.patient,
+            fee: appointment.doctor.fee,
+            confirmed: appointment.confirmed,
+            card_number: cardNo
+        };
+
+        Payment|http:ClientError payment =
+                hospitalServicesEP->/healthcare/payments.post(paymentSettlement);
+
+        if payment !is Payment {
+            log:printError("Payment settlement failed", payment);
+            if payment is http:ClientRequestError {
+                return <http:NotFound> {body: string `unknown hospital, patient, or category`};
+            }
+            return <http:InternalServerError> {body: payment.message()};
+        }
+
+        ReservationResponse reservationResponse = {
+            appointmentNo: appointmentNumber,
+            doctorName: reservationRequest.doctor,
+            patient: reservationRequest.patient.name,
+            actualFee: payment.actualFee,
+            discount: payment.discount,
+            discounted: payment.discounted,
+            paymentID: payment.paymentID,
+            status: payment.status
+        };
+
+        string[] messageContent = from var [key, value] in reservationResponse.entries()
+            let string str = key + ": " + value.toString()
+            select str;
+
+        string emailBody = string:'join("\n", ...messageContent);
+
+        email:Message email = {
+            to: reservationRequest.patient.email,
+            subject: "Payment Status",
+            body: emailBody
+        };
+
+        email:Error? sendMessage = smtpClient->sendMessage(email);
+        if sendMessage is email:Error {
+            return <http:InternalServerError> {body: sendMessage.message()};
+        }
+        log:printDebug("Email sent successfully",
+                       status = "Payment Status",
+                       body = emailBody);
```

Comment:
Are these really required? Recipient name may make more sense (something that changes from request to request).

Reply:
Wanted to make the debug log a bit more detailed. That's why I added them.

Reply:
A debug log should basically contain sufficient information to troubleshoot an issue. I think hardcoded  `status = "Payment Status"` doesn't add much value, we can incorporate it in the mail itself. On the other hand, including the entire email body may be too much content, it would be better to include only the uniquely identifying parts from it.

Reply:
That makes sense. I'll fix it